src/main/scala/exercises/06_Collections.scala

```diff
@@ -129,67 +160,48 @@ object Collections {
    * Implement it whatever way suites you best. Hint: it can be done in a 
    * neat way using recursion. 
    */
-  def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
-  }
+  def tails[T](l: List[T]): List[List[T]] = 
+    if (l.isEmpty) List(l) 
+    else l::tails(l.tail)
   
   /**
    * Find the maximum element in a list, e.g. maxElementInList(List(1,9,3,5)) == 9
    * 
    * As usual, various ways exist: pattern matching, folding, ...
    */
-  def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
-  }
+  def maxElementInList(l: List[Int]): Int = l.max
 
   /**
    * Calculate the sum of the equally position elements
    * of the two list
    */
+  
   def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = {
-    error("fix me")
+  // This was Prof Ben's solution -- I tried many different things that 
+  // ended up not working out.
+  // I tried doing a for loop and adding every nth element from 
+  // l2 to the nth element in l1... and am not sure what went wrong.
```

Comment:
Same as above comment. If you want an imperative solution, you need to make you are updating `l1` every time.